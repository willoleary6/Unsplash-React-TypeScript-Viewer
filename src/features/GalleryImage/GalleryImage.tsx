import React, { useState } from "react";
import { LoadingSpinner } from "../LoadingSpinner";

interface GalleryImageProps {
    sourceUrl: string;
}

export const GalleryImage = ({ sourceUrl }: GalleryImageProps): JSX.Element => {
    const [loaded, setLoaded] = useState(false);
    return (
        <>
            {loaded ? null : <LoadingSpinner />}
            <div className={loaded ? "" : "hidden"}>
                <img
                    alt="gallery"
                    className="block p-1 md:p-2 object-cover object-center w-full h-full rounded-lg"
                    src={sourceUrl}
                    onLoad={() => setLoaded(true)}
                />
            </div>
        </>
    );
};
