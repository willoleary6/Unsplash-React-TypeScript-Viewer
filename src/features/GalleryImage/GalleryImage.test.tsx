import React from "react";
import { GalleryImage } from "./GalleryImage";

import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

describe("GalleryImage render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        const wrapper = shallow(
            <GalleryImage sourceUrl="https://images.unsplash.com/photo-1648737963059-59ec8e2d50c5?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzMjU1NzJ8MXwxfHNlYXJjaHwxfHxvZmZpY2V8ZW58MHx8fHwxNjUxNjA5NTQ4&ixlib=rb-1.2.1&q=85" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
