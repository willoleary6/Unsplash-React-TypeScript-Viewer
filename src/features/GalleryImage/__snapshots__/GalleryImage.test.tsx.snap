// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryImage render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<Fragment>
  <LoadingSpinner />
  <div
    className="hidden"
  >
    <img
      alt="gallery"
      className="block p-1 md:p-2 object-cover object-center w-full h-full rounded-lg"
      onLoad={[Function]}
      src="https://images.unsplash.com/photo-1648737963059-59ec8e2d50c5?crop=entropy&cs=srgb&fm=jpg&ixid=MnwzMjU1NzJ8MXwxfHNlYXJjaHwxfHxvZmZpY2V8ZW58MHx8fHwxNjUxNjA5NTQ4&ixlib=rb-1.2.1&q=85"
    />
  </div>
</Fragment>
`;
